src/CopyOnWrite/Copy.cs

```diff
@@ -976,7 +976,8 @@ private static int ParseIntFromEnvironmentVariableOrDefault(string environmentVa
         }
 
         #region CopyOnWrite functionality
-        private static readonly ICopyOnWriteFilesystem _cow = CopyOnWriteFilesystemFactory.GetInstance();
+        private static readonly ICopyOnWriteFilesystem CoW = CopyOnWriteFilesystemFactory.GetInstance();
+        private static readonly bool DisableCoW = Environment.GetEnvironmentVariable("MSBUILDDISABLECOPYONWRITE") is not null;
```

Comment:
This is already toggled at the task level:
https://github.com/microsoft/MSBuildSdks/blob/c8f848540db223df28b3343588111592c9eb2e8a/src/CopyOnWrite/build/Microsoft.Build.CopyOnWrite.targets#L5C14-L5C14

I probably wouldn't do both. If you do want this, you should also probably check the value so `MSBUILDDISABLECOPYONWRITE=0` doesn't disable it, for example.

Reply:
WIll remove here. There's another copy of this code in Artifacts, I'll either remove and add a flag to the targets file, or keep and improve the comparison

Reply:
I think it's the right approach for artifacts though. For Copy, if you opt out it should fallback to the old task entirely not use the "new" task with the feature disabled.

Although there should probably be a single way to opt-out for both, But to do that you would have to pass in the property to the artifacts Task.

Reply:
Added param to the Robocopy task code and updated RobocopyFiles and CopyArtifacts targets to use EnableCopyOnWriteWin